contracts/zaps/OlympusZapV2.sol

```diff
@@ -0,0 +1,293 @@
+                    IBondDepoV2(depo).bondPrice(bid),
+                feo
+            );
+            // emit zapIn
+            emit zapIn(msg.sender, toToken, OHMRec, affiliate);
+        } else {
+            require(toToken == sOHM || toToken == gOHM, "toToken must be sOHM or gOHM");
+            uint256 toInvest = _pullTokens(fromToken, amountIn, affiliate, true);
+            uint256 tokensBought = _fillQuote(fromToken, OHM, toInvest, swapTarget, swapData);
+            OHMRec = _enterOlympus(tokensBought, toToken);
+            require(OHMRec > minToToken, "High Slippage");
+            emit zapIn(msg.sender, sOHM, OHMRec, affiliate);
+        }
+    }
+
+    /// @notice This function withdraws assets from OlympusDAO, receiving tokens or ETH
+    /// @param fromToken The ibToken being withdrawn
+    /// @param amountIn The quantity of fromToken to withdraw
+    /// @param toToken Address of the token to receive (0 address if ETH)
+    /// @param minToTokens The minimum acceptable quantity of tokens to receive. Reverts otherwise
+    /// @param swapTarget Excecution target for the swap or zap
+    /// @param swapData DEX or Zap data
+    /// @param affiliate Affiliate address
+    /// @return tokensRec Quantity of aTokens received
+    function ZapOut(
+        address fromToken,
+        uint256 amountIn,
+        address toToken,
+        uint256 minToTokens,
+        address swapTarget,
+        bytes calldata swapData,
+        address affiliate
+    ) external stopInEmergency returns (uint256 tokensRec) {
+        // make sure from token is not sOHM or gOHM
+        require(fromToken == sOHM || fromToken == gOHM, "fromToken must be sOHM or gOHM");
+        // pull users tokens and store amount in
+        amountIn = _pullTokens(fromToken, amountIn);
+
+        uint256 OHMRec = _exitOlympus(fromToken, amountIn);
+
+        tokensRec = _fillQuote(OHM, toToken, OHMRec, swapTarget, swapData);
+
+        require(tokensRec >= minToTokens, "High Slippage");
+
+        uint256 totalGoodwillPortion;
+
+        if (toToken == address(0)) {
+            totalGoodwillPortion = _subtractGoodwill(ETHAddress, tokensRec, affiliate, true);
+            payable(msg.sender).transfer(tokensRec - totalGoodwillPortion);
+        } else {
+            totalGoodwillPortion = _subtractGoodwill(toToken, tokensRec, affiliate, true);
+            IERC20(toToken).safeTransfer(msg.sender, tokensRec - totalGoodwillPortion);
+        }
+        tokensRec = tokensRec - totalGoodwillPortion;
+        emit zapOut(msg.sender, toToken, tokensRec, affiliate);
+    }
+
+    ////////////////////////// INTERNAL //////////////////////////
+
+    function _enterOlympus(uint256 amount, address toToken) internal returns (uint256) {
+        if (toToken == gOHM) {
+            IStaking(staking).stake(address(this), amount, false, false);
+            IStaking(staking).claim(address(this), false);
+            uint256 gOHMRec = IStaking(staking).wrap(msg.sender, amount);
```

Comment:
same thing here. don't need to wrap. stake takes in a `rebase` boolean:
https://github.com/OlympusDAO/olympus-contracts/blob/main/contracts/Staking.sol#L252-L264

Reply:
removed, and fixed params